Updater failure due to using newer Android APIs

Updater fails to check for updates due to reflecting an API that doesn't exist on Android 7.


Logcat from Pixel in the Discord:
```
[2026-04-28 15:05:23.243 Uid(value=10572):3231:3361 E/Discord]
java.lang.Error: java.lang.NoSuchFieldException: No field path in class Ldalvik/system/DexPathList$Element; (declaration of 'dalvik.system.DexPathList$Element' appears in /system/framework/core-libart.jar)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java, [Aliucord] Uncaught exception on thread pool-25-thread-3:1139)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
Caused by: java.lang.NoSuchFieldException: No field path in class Ldalvik/system/DexPathList$Element; (declaration of 'dalvik.system.DexPathList$Element' appears in /system/framework/core-libart.jar)
	at java.lang.Class.getDeclaredField(Native Method)
	at com.aliucord.utils.ReflectUtils.getField(ReflectUtils.java:176)
	at com.aliucord.utils.ReflectUtils.getField(ReflectUtils.java:159)
	at com.aliucord.updater.CoreUpdater._customCoreLoaded_delegate$lambda$0(CoreUpdater.kt:164)
	at com.aliucord.updater.CoreUpdater.$r8$lamb

[2026-04-28 15:05:23.243 Uid(value=10572):3231:3361 E/Discord]
da$VOFGIbCEbVlLZrvMmZnFVYlFU_c(CoreUpdater.kt)
	at com.aliucord.updater.CoreUpdater$$ExternalSyntheticLambda2.invoke(D8$$SyntheticClass:0)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.aliucord.updater.CoreUpdater.get_customCoreLoaded(CoreUpdater.kt:150)
	at com.aliucord.updater.CoreUpdater.isCustomCoreLoaded(CoreUpdater.kt:147)
	at com.aliucord.updater.CoreUpdater.checkForUpdates(CoreUpdater.kt:37)
	at com.aliucord.updater.PluginUpdater.checkUpdates(PluginUpdater.java:53)
	at com.aliucord.Main.lambda$startAllPlugins$22(Main.java:543)
	at com.aliucord.Main$$ExternalSyntheticLambda33.run(D8$$SyntheticClass:0)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	... 2 more
```

--- a/Aliucord/src/main/java/com/aliucord/updater/CoreUpdater.kt
+++ b/Aliucord/src/main/java/com/aliucord/updater/CoreUpdater.kt
@@ -154,15 +154,14 @@ internal object CoreUpdater {
             "dexElements",
         )!! as Array<Any>
 
+        val fieldName = when {
+            Build.VERSION.SDK_INT < 23 -> "file"
+            Build.VERSION.SDK_INT < 26 -> "zip"
+            else -> "path"
+        }
+
         dexElements.any { element ->
-            val fieldName = when {
-                Build.VERSION.SDK_INT < 23 -> "file"
-                Build.VERSION.SDK_INT < 26 -> "zip"
-                else -> "path"
-            }
-
             val file = ReflectUtils.getField(element, fieldName)!! as File
-
             file.name == "Aliucord.custom.zip"
         }
     }

--- a/Aliucord/src/main/java/com/aliucord/updater/CoreUpdater.kt
+++ b/Aliucord/src/main/java/com/aliucord/updater/CoreUpdater.kt
@@ -2,6 +2,7 @@ package com.aliucord.updater
 
 import android.content.ActivityNotFoundException
 import android.content.Intent
+import android.os.Build
 import com.aliucord.*
 import com.aliucord.api.NotificationsAPI
 import com.aliucord.entities.NotificationData
@@ -153,11 +154,17 @@ internal object CoreUpdater {
             "dexElements",
         )!! as Array<Any>
 
-        dexElements.findLast { element ->
-            val path = ReflectUtils.getField(element, "path")!! as File
+        dexElements.any { element ->
+            val fieldName = when {
+                Build.VERSION.SDK_INT < 23 -> "file"
+                Build.VERSION.SDK_INT < 26 -> "zip"
+                else -> "path"
+            }
 
-            path.name == "Aliucord.custom.zip"
-        } != null
+            val file = ReflectUtils.getField(element, fieldName)!! as File
+
+            file.name == "Aliucord.custom.zip"
+        }
     }
 
     /**

